### PR TITLE
Upgrade to Next.js 10

### PR DESCRIPTION
**Pull request by kidonng**
This is mostly the same as #176, except it now correctly fixes `components/Function.tsx` (`Reduce of empty array with no initial value`) to avoid problems like https://github.com/denoland/deno_website2/issues/1726#issuecomment-814667414.

You can confirm that it doesn't have the extra `, ` in function parameters, and function without parameters (like `Deno.cwd()` renders fine.

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,42 +1,8 @@
-const withPrefresh = require("@prefresh/next");
+const withPreact = require("next-plugin-preact");
 
-module.exports = withPrefresh({
+module.exports = withPreact({
```
Comment (kt3k) [inline next.config.js:3]: Is this change necessary for updating to next.js 10?

**Review by kt3k — APPROVED** [review]
This seems fixing the previous issue (unnecessary comma in function args). (Confirmed at https://doc-website-ll2b7ey1p-denoland.vercel.app/builtin/stable )

LGTM.

File: next.config.js (hunk 1 above)
Comment (kidonng) [reply next.config.js:3]: It's taken from https://github.com/denoland/deno_website2. I think upgrading preact together should make the upgrade less problematic.
